src/main/java/org/openhds/resource/controller/census/IndividualRestController.java

```diff
@@ -188,6 +216,143 @@ protected Individual register(IndividualHouseholdRegistration registration, Stri
+                filteredRelationships.add(relationship);
+            }
+        }
+        return filteredRelationships;
+    }
+
+
+    private class EventStructure {
+        public List<InMigration> inMigrations;
+        public List<OutMigration> outMigrations;
+        public List<Death> deaths;
+        public List<PregnancyObservation> pregnancyObservations;
+        public List<PregnancyOutcome> pregnancyOutcomes;
+
+        // male
+        private EventStructure(List<InMigration> inMigrations, List<OutMigration> outMigrations,
+                               List<Death> deaths){
+            this.inMigrations = inMigrations;
+            this.outMigrations = outMigrations;
+            this.deaths = deaths;
+        }
+
+        // female
+        private EventStructure(List<InMigration> inMigrations, List<OutMigration> outMigrations,
+                               List<Death> deaths, List<PregnancyObservation> pregnancyObservations,
+                               List<PregnancyOutcome> pregnancyOutcomes ){
+            this.inMigrations = inMigrations;
+            this.outMigrations = outMigrations;
+            this.deaths = deaths;
+            this.pregnancyObservations = pregnancyObservations;
+            this.pregnancyOutcomes = pregnancyOutcomes;
+
+        }
+
+    }
+
+
+    @RequestMapping(value = "/getEvents", method = RequestMethod.GET)
+    public EventStructure getEvents(@RequestParam String individualUuid) {
+
+        Individual indiv = individualService.findOne(individualUuid);
+
+        // InMigrations
+        EntityIterator<InMigration> inMigrations = inMigrationService.findAll(new Sort("uuid"));
+        List<InMigration> filteredInMigrations = new ArrayList<>();
+        for (InMigration inMigration: inMigrations) {
+            if(     inMigration.getIndividual().getUuid().equals(individualUuid) ) {
+                filteredInMigrations.add(inMigration);
+            }
+        }
+
+        // OutMigrations
+        EntityIterator<OutMigration> outMigrations = outMigrationService.findAll(new Sort("uuid"));
+        List<OutMigration> filteredOutMigrations = new ArrayList<>();
+        for (OutMigration outMigration: outMigrations) {
+            if(     outMigration.getIndividual().getUuid().equals(individualUuid) ) {
+                filteredOutMigrations.add(outMigration);
+            }
+        }
+
+        // Deaths
+        EntityIterator<Death> deaths = deathService.findAll(new Sort("uuid"));
+        List<Death> filteredDeaths = new ArrayList<>();
+        for (Death death: deaths) {
+            if(     death.getIndividual().getUuid().equals(individualUuid) ) {
+                filteredDeaths.add(death);
+            }
+        }
+
+        if(indiv.getGender().equals(("MALE"))){
+            EventStructure events = new EventStructure(filteredInMigrations, filteredOutMigrations, filteredDeaths);
+            return events;
+
+        }
+        else{
```

Comment:
This list building structure is repeated several times. Can it be parameterized to reduce duplication?